scripts/generate_python_types.py

```diff
@@ -0,0 +1,303 @@
+            class_name = f"{containing_class}_{field_name}"
+            class_code, inner_types_code = generate_class(class_name, type_field)
+
+            if not inner_types_code:
+                inner_types_code = io.StringIO()
+
+            inner_types_code.write("\n\n")
+            inner_types_code.write(class_code)
+
+            return class_name, inner_types_code
+
+    else:
+        raise ValueError(f"Invalid type field: {type_field}")
+
+
+def _parse_composite_type(
+    containing_class: str, field_name: str, type_dict: dict[str, Any]
+) -> tuple[str, io.StringIO | None]:
+    assert isinstance(type_dict["type"], str)
+
+    type_name, type_code = None, None
+    if "subType" in type_dict:
+        try:
+            type_name, type_code = parse_type(
+                containing_class, field_name, type_dict["subType"]
+            )
+        except Exception as e:
+            e.add_note(f"Error in subType {type_dict["subType"]}")
+            raise e
+
+    match type_dict["type"]:
+        case "attrsOf":
+            return f"dict[str, {type_name}]", type_code
+        case "listOf":
+            return f"list[{type_name}]", type_code
+        case "nullOr":
+            return f"None | {type_name}", type_code
+        case "oneOf":
+            type_code = io.StringIO()
+            type_names = []
+            for sub_type in type_dict["types"]:
+                try:
+                    sub_type_name, sub_type_code = parse_type(
+                        containing_class, field_name, sub_type
+                    )
+                except Exception as e:
+                    e.add_note(f"Error in subType {sub_type}")
+                    raise e
+
+                type_names.append(sub_type_name)
+                if sub_type_code:
+                    type_code.write(sub_type_code.getvalue())
+
+            # Can't use | syntax in all cases, Union always works
+            return f'Union[{", ".join(type_names)}]', type_code
+        case "enum":
+            return (
+                f'Literal[{", ".join(f"{repr(value)}" for value in type_dict["choices"])}]',
+                None,
+            )
+        case _:
+            return _parse_simple_type(type_dict["type"]), None
+
+
+def _parse_simple_type(type_str: str) -> str:
+    match type_str:
+        case "str":
+            return "str"
+        case "absolute-pathname":
+            return "str"
+        case "bool":
+            return "bool"
+        case "int":
+            return "int"
+        case "anything":
+            return "Any"
+        # Set up discriminated unions to reduce error messages when validation fails
+        case "deviceType":
+            return '"deviceType" = Field(..., discriminator="type")'
+        case "partitionType":
+            return '"partitionType" = Field(..., discriminator="type")'
+        case _:
+            # Probably a type alias, needs to be quoted in case the type is defined later
+            return f'"{type_str}"'
```

Comment:
With traditional Python you would write it like that:

```suggestion
    types_map = {
        "str": "str",
        "absolute-pathname": "str",
        "bool": "bool",
        "int": "int",
        "anything": "Any",
        # Set up discriminated unions to reduce error messages when validation fails
        "deviceType": '"deviceType" = Field(..., discriminator="type")',
        "partitionType": '"partitionType" = Field(..., discriminator="type")',
    }
    try:
        return types_map[type_str]
    except KeyError:
        # Probably a type alias, needs to be quoted in case the type is defined later
        return f'"{type_str}"'
```

This is (still) the pythonic way. I understand that `match/switch/case` is known in other languages.

Reply:
True, it's more terse anyway. I used match here initially because I called a function in one of the branches, but when all the results are static, that's not needed anymore.